gst/nnstreamer/tensor_filter/tensor_filter_common.c

```diff
@@ -1934,3 +1934,36 @@ accl_hw_get_type (void)
 
   return g_accl_hw_type_id__volatile;
 }
+
+/**
+ * @brief check if the given hw is supported by the framework
+ */
+gboolean
+gst_tensor_filter_check_hw_availability (const GstTensorFilterFramework * fw,
+    accl_hw hw)
+{
+  gint ret, idx = 0;
+  gboolean available = FALSE;
+  GstTensorFilterFrameworkInfo info;
+  GstTensorFilterProperties prop;
+
+  if (GST_TF_FW_V0 (fw) &&
+      (!fw->checkAvailability || fw->checkAvailability (hw) == 0)) {
+    available = TRUE;
+  } else if (GST_TF_FW_V1 (fw)) {
+    gst_tensor_filter_properties_init (&prop);
+    ret = fw->getFrameworkInfo (fw, &prop, NULL, &info);
+    if (ret == 0) {
+      for (idx = 0; idx < info.num_hw; idx++) {
+        if (info.hw_list[idx] == hw) {
+          available = TRUE;
+          break;
+        }
+      }
+    } else if (ret == -ENOENT) {
+      available = TRUE;
```

Comment:
Should this be `available = FALSE`?

Reply:
aah, yes. `available` should be `FALSE` for all `ret < 0`.